cmd/kube-apiserver/app/server.go

```diff
@@ -686,6 +688,11 @@ func Complete(s *options.ServerRunOptions) (completedServerRunOptions, error) {
 				s.Authentication.ServiceAccounts.MaxExpiration > upBound {
 				return options, fmt.Errorf("the serviceaccount max expiration must be between 1 hour to 2^32 seconds")
 			}
+			// When service-account-extend-token-expiration is enabled during transition time, max expiration must be set
+			// long enough for fixing clients.
+			if s.Authentication.ServiceAccounts.ExtendExpiration && s.Authentication.ServiceAccounts.MaxExpiration < serviceaccount.ExpirationExtensionSeconds*time.Second {
```

Comment:
Don't check MaxExpiration here. It should only bound non-special tokens.

Reply:
There was a converstation between Jordan and me.

>  @liggitt
> Jordan Liggitt 19 days ago Member
> I'd expect this to be bounded to the lesser of --service-account-max-token-expiration or 1 year... doing the computation in TokenREST#Create as suggested in https://github.com/kubernetes/kubernetes/pull/89549/files#r399304171 would give you access to the maxExpirationSeconds
> 
>  @happinesstaker
> JIAJIE YANG 19 days ago Author
> I wonder what people usually set for this flag.
> GKE kind of set this to 48h, which makes me feel like it might not be a good idea setting our extended expiration capped by this.
> 
>  @liggitt
> Jordan Liggitt 16 days ago Member
> The purpose of the max-expiration flag is to let the admin know when it is safe to rotate the signing key. If we use that key to issue tokens valid for longer than the max expiration, it no longer serves that purpose. I would also accept erroring on startup if extend-expiration was true and max-expiration was less than one year
> 
>  @happinesstaker
> JIAJIE YANG 15 days ago Author
> I would be leaning towards erroring if extend-expiration was enabled but max-expiration was set to a short time, instead of silently make the token expire in a shorter time when admin was not aware these two flags are related.

Reply:
Seems that we set service-account-max-token-expiration=48h in GKE by default -> checking this always results in error @liggitt 
maybe we do not cap these special tokens?

Reply:
It doesn't seem great for this mechanism to enable obtaining tokens with longer lifetimes than could be directly requested. That seems likely to lead to people requesting the magic duration for that purpose.

Reply:
Hmm... I guess requiring max-token-expiration to be as long as the extended period would allow requesting/obtaining tokens that were valid for that long, which would not trigger warnings, and shortening the max-token-expiration would then become unsafe.

Is there a way to make it clearer to an admin that sets `--service-account-max-token-expiration=48h --service-account-extend-token-expiration` that it is not safe to rotate their signing key until the extended expiration passes?

Reply:
I guess we can just document the interaction, and log a warning if `service-account-max-token-expiration` is shorter than the extended expiration.

Reply:
Done. Changed to warning, and document the interaction in the flag help